game-core/src/main/java/games/strategy/engine/framework/startup/launcher/LocalLauncher.java

```diff
@@ -63,7 +64,9 @@ public void launch() {
       final Set<Player> gamePlayers =
           gameData
               .getGameLoader()
-              .newPlayers(playerListing.getLocalPlayerTypeMap(new PlayerTypes()));
+              .newPlayers(
+                  playerListing.getLocalPlayerTypeMap(
+                      new PlayerTypes(Injections.getInstance().getPlayerTypes())));
```

Comment:
I think we need to try and avoid static coupling. That would mean to inject the 'playerTypes' as a constructor arg. This way only code that is instantiating the `LocalLauncher` is aware of `Injections`, but the code using the injected values is not aware of how it is injected.

I have not researched what this would involve, this is a bit of a quick comment. Let me know what you think and if there are any serious issues with constructor injection.

Reply:
I looked at passing around Injections and that is a massive change to the code base.  It would touch a lot of places.  I really don't want to do that as part of this PR.

Reply:
Injections should not be passed around. I do object to Injections being used in a static context (the javadoc on the class says to avoid using it like that).

When an object uses Injections, it gets any data from Injections constructor injected. It should be no more of a massive change than what is already done. Instead of accessing 'Injections.getInsance().getPlayerTypes()' directly, pass the data to the constructor and it just becomes a constructor arg. Let me know if I'm missing something..

Reply:
`Injections should not be passed around` and `object to Injections being used in a static context` are conflicting ideas to me.  If Injections aren't passed around, then you have to access them through a static context.  Can you explain how you access the Injections without passing them around and without accessing them through a static context?

Reply:
Example: `new MyObject(Injections.getInstance().getData());`


With:
```
@AllArgsConstructor
class MyObject {
  private final Data data;

    void foo() {
             data.doSomething();
     }
}
```


vs:

```
class MyObject {

    void foo() {
             Injections.getInstance().getData().doSomething();
     }
}
```

Hopefully that clarifies through an example. A thing to note is in the first example the implementation of `MyObject` is not aware of Injections at all. Once we have that pattern, it is only a matter of keeping object creation clean (and paves the way to have the option of a clean bona fide IOC framework being dropped in to cleanly replace `Injections`)

Reply:
That's what I thought you meant.  And my concern is still valid: `I looked at passing around Injections and that is a massive change to the code base.`.  I would still have to pass Injections class to all of the callers before `new PlayerTypes` is called.  That is a large change to the codebase.  There are a lot of callers and I'd be touching a good portion of the code path.  I don't think that fits in this PR.

Reply:
To give a brief example, this specific code is in `LocalLauncher#loadGame`.  I would have to modify `LocalLauncher#loadGame` to accept an `Injections` or a collection of `PlayerTypes`.  I would then need to modify `LocalLauncher#launch` to accept one or the other.  Since this is an `Override` method, I'd have to modify the parent method as well and any other `Override` versions.  I would then need to modify `MainPanelBuilder#buildMainPanel` to accept one or the other so that I could pass it into `LocalLaunch#launch`.  `MainPanelBuilder#buildMainPanel` is called from `MainFrame#buildMainFrame` so that would need to be modified.  And then its caller, `GameRunner#start` would also need to be modified.  The caller of `GameRunner#start` is `HeadedGameRunner#main` which is where `Injections` is initialized.

So, just for this one place where `Injections.getPlayerTypes()` is used, I'd need to modify at least 4 different classes as well as one parent class and any other child classes of it.  `Injections.getPlayerTypes()` is used 5 times so that is even more classes I'd have to modify.

Reply:
That list is actually not unreasonable  :grin: 
I'll work on this next step to help out.